tdrs-frontend/src/components/UploadReport/UploadReport.jsx

```diff
@@ -38,6 +42,10 @@ function UploadReport({ handleCancel, header, stt }) {
     headerRef.current.focus()
   }, [])
 
+  useEffect(() => {
+    dispatch(getAvailableFileList({ year: selectedYear }))
+  }, [dispatch, getAvailableFileList, selectedYear])
```

Comment:
There is a React warning in the console suggesting `getAvailableFileList` is unnecessary here.

Reply:
```suggestion
  }, [dispatch, selectedYear])
```
The optional array passed in to `useEffect` indicates values that must change in order for this to be re-rendered:

https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Since `getAvailableFileList` is from a scope outside this file it is not a valid dependency here.